services/httpd/handler.go

```diff
@@ -1873,6 +1875,39 @@ func requestID(inner http.Handler) http.Handler {
 	})
 }
 
+func (h *Handler) SetHeadersHandler(handler http.Handler) http.Handler {
+	return http.HandlerFunc(h.SetHeadersWrapper(handler.ServeHTTP))
+}
+
+// wrapper that adds user supplied headers to the response.
+func (h *Handler) SetHeadersWrapper(f func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
+	if len(h.Config.HTTPHeaders) == 0 {
+		return f
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("attempting to set headers for request")
+		for _, kvp := range h.Config.HTTPHeaders {
+			if len(kvp) != 2 {
+				// NOTE(ayan): maybe this should be an error.  maybe we should ensure
```

Comment:
Will fix this comment -- will use standard case and edit the comment to be more concise.